pkg/api/dashboard.go

```diff
@@ -350,6 +359,144 @@ func (hs *HTTPServer) deleteDashboard(c *contextmodel.ReqContext) response.Respo
 	})
 }
 
+// DeleteDashboardsByTagsUIDs swagger:route DELETE /dashboards dashboards deleteDashboardsByTagsORUIDs
+//
+// Delete dashboard by tags and UIDs.
+//
+// Will delete the dashboard given the specified tags.
+//
+// Responses:
+// 207: deleteDashboardsByTagsORUIDsResponse
+// 200: deleteDashboardsByTagsORUIDsResponse
+// 401: unauthorisedError
+// 403: forbiddenError
+// 404: notFoundError
+// 500: internalServerError
+func (hs *HTTPServer) DeleteDashboardsByTagsORUIDs(c *contextmodel.ReqContext) response.Response {
+	return hs.deleteDashboardsByTagsORUIDs(c)
+}
+
+func (hs *HTTPServer) deleteDashboardsByTagsORUIDs(c *contextmodel.ReqContext) response.Response {
+	deleteDashReq := dashboards.DeleteDashboardsRequest{}
+	if err := web.Bind(c.Req, &deleteDashReq); err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
+	}
+
+	if len(deleteDashReq.Tags) == 0 && len(deleteDashReq.UIDs) == 0 {
+		return response.Error(http.StatusBadRequest, "Provide either tags or UIDs to delete dashboards", errors.New("unexpected input"))
+	}
+	if len(deleteDashReq.Tags) != 0 && len(deleteDashReq.UIDs) != 0 {
+		return response.Error(http.StatusBadRequest, "Both tags and uids are not allowed to delete dashboards", errors.New("unexpected input"))
+	}
+
+	deleteDashboards, rsp := hs.getDashboardsHelper(c.Req.Context(), c.OrgID, deleteDashReq.Tags, deleteDashReq.UIDs, c.SignedInUser)
+	if rsp != nil {
+		return rsp
+	}
+
+	totalCount := len(deleteDashboards)
+	finalRes := dashboards.DeleteDashboardsResponse{}
+	delDashRes := make(chan dashboards.DeleteDashboardStatus, totalCount)
+
+	for _, dash := range deleteDashboards {
+		go func(dash dashboards.DashboardSearchProjection) {
+			delDashRes <- hs.deleteDashboardByTagORUID(c, dash)
+		}(dash)
+	}
+
+	for i := 0; i < totalCount; i++ {
+		res := <-delDashRes
+		finalRes.Status = append(finalRes.Status, res)
+		if res.StatusCode == http.StatusOK {
+			finalRes.SuccessCount += 1
+		} else {
+			finalRes.FailedCount += 1
+		}
+	}
+
+	if totalCount == finalRes.SuccessCount {
+		return response.JSON(http.StatusOK, finalRes)
+	}
+	if totalCount == finalRes.FailedCount {
+		return response.JSON(http.StatusForbidden, finalRes)
+	}
+
+	return response.JSON(http.StatusMultiStatus, finalRes)
+
+}
+
+func (hs *HTTPServer) deleteDashboardByTagORUID(c *contextmodel.ReqContext, dash dashboards.DashboardSearchProjection) dashboards.DeleteDashboardStatus {
+
+	dashboard := &dashboards.Dashboard{ID: dash.ID, UID: dash.UID, Slug: dash.Slug, OrgID: c.OrgID, FolderID: dash.FolderID, IsFolder: dash.IsFolder}
```

Comment:
Can we refactor it into a more readable form:
`{`
                                            `ID: dash.ID,`
                                            `UID: dash.UID,`
                                            `Slug: dash.Slug,`
                                            `...`
                                         `}`

Reply:
Done